src/Board.cpp

```diff
@@ -0,0 +1,226 @@
+    if (isKingUnderAttack(m_turn == PieceColor::WHITE ? PieceColor::BLACK : PieceColor::WHITE))
+        ret = 41;
+    
+    else ret = 42;
+
+    //switching turn
+    m_turn = m_turn == PieceColor::WHITE ? PieceColor::BLACK : PieceColor::WHITE;
+    
+    return ret;
+
+}
+
+/**
+ * @brief Checks if the path between the source and destination positions is clear (i.e., unoccupied).
+ * @param src The source position.
+ * @param dst The destination position.
+ * @return true if the path is clear, false otherwise.
+ */
+bool Board::isPathClear(const Place& src, const Place& dst) const
+{
+    int dx = dst.m_col - src.m_col;
+    int dy = dst.m_row - src.m_row;
+
+    // calculate the direction of the move
+    int dxSign = dx == 0 ? 0 : (dx > 0 ? 1 : -1);
+    int dySign = dy == 0 ? 0 : (dy > 0 ? 1 : -1);
+
+    // check if the move is diagonal or horizontal/vertical
+    bool diagonal = abs(dx) == abs(dy);
+    bool straight = dx == 0 || dy == 0;
+
+    if (!diagonal && !straight)
+        return false;  // invalid move
+
+    // check if there are pieces in the way
+    int steps = diagonal ? abs(dx) : dx != 0 ? abs(dx) : abs(dy);
+
+    Place curr = src;
+    for (int i = 1; i < steps; i++)
+    {
+        curr.m_row += dySign;
+        curr.m_col += dxSign;
+        auto it = std::find_if(m_board.begin(), m_board.end(),
+            [&](const auto& p) {return p.second == curr; });
+        if (it != m_board.end())
+            return false;  // there is a piece in the way
+    }
+
+    return true;  // path is clear
+}
+
+
+/**
+ * @brief Validates the given command string.
+ * @param input The command string to validate.
+ * @return true if the command string is valid, false otherwise.
+ */
+bool Board::isValidInput(const std::string& input) const {
+    if (input.length() != 4)
+        return false;
+    
+    if (input[0] < 'a' || input[0] > 'h' || input[2] < 'a' || input[2] > 'h')
+        return false;
+    
+    if (input[1] < '1' || input[1] > '8' || input[3] < '1' || input[3] > '8')
+        return false;
+   
+    return true;
+}
+
+
+/**
+ * @brief Moves a piece from the source to the destination position.
+ * @param src The source position.
+ * @param dst The destination position.
+ */
+void Board::movePiece(const Place& src, const Place& dst) {
+
+    auto itSrc = std::find_if(m_board.begin(), m_board.end(),
+        [&](const auto& p) { return p.second == src; });
+    auto itDst = std::find_if(m_board.begin(), m_board.end(),
+        [&](const auto& p) { return p.second == dst; });
+
+    itSrc->second = dst;
+    if (itDst != m_board.end())
+        m_board.erase(itDst);
+}
+
+/**
+ * @brief Checks if the king of the given color is under attack.
+ * @param color The color of the king to check.
+ * @return true if the king is under attack, false otherwise.
+ */
+bool Board::isKingUnderAttack(const PieceColor& color) const {
+    //getting the place of the king
+    Place kingPlace;
+    for (const auto& p : m_board) {

```

Comment:
Example for making code more readable - this piece of self-contained logic should be its own private method, called something meaningful like "getKingPosition()". That way the reader knows what your function does without needing to read it and understand its implementation